aws/resource_aws_route.go

```diff
@@ -250,27 +254,33 @@ func resourceAwsRouteCreate(d *schema.ResourceData, meta interface{}) error {
 
 	if v, ok := d.GetOk("destination_ipv6_cidr_block"); ok {
 		err = resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-			route, err = findResourceRoute(conn, d.Get("route_table_id").(string), "", v.(string))
+			route, err = resourceAwsRouteFindRoute(conn, d.Get("route_table_id").(string), "", v.(string))
 			return resource.RetryableError(err)
 		})
 		if err != nil {
 			return fmt.Errorf("Error finding route after creating it: %s", err)
 		}
 	}
 
-	d.SetId(routeIDHash(d, route))
+	d.SetId(resourceAwsRouteID(d, route))
 	resourceAwsRouteSetResourceData(d, route)
 	return nil
 }
 
 func resourceAwsRouteRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+
+	err := resourceAwsRouteCheckForImport(d)
```

Comment:
The `Importer.State` and `Create` functions should handle ID errors and setting the appropriate Terraform state required for read unless we switch to using `schema.ImportStatePassthrough` for `Importer.State`. The same applies to other usage in `Update`/`Delete`/`Exists`.